### DISCUSSION TITLE
Access `Design.IsDesignMode` in xaml

### DISCUSSION THREAD

**Discussion by penenkel** [AvaloniaUI/Avalonia#6588] [Q&A] [answered]
I want to toogle a the visibility of a control in xaml based on `Design.IsDesignMode` without having to add a corresponding property to my ViewModel.

As `Design.IsDesignMode` is a static property, we can reference it from xaml via `x:Static`. Unfortunately we cannot negate it. 
In the following example, the second one will obviously not work because such a negation is only supported for bindings.
```
    <TextBlock IsVisible="{x:Static Design.IsDesignMode}"  Text="We are in DesignMode" />
    <TextBlock IsVisible="{x:Static !Design.IsDesignMode}"  Text="We are not in DesignMode" />
```

I also considered using bindings, but as far as I know we cannot (yet) bind to static properties; see #3815. So this will also not work. (Also I`m not sure if this would be good a syntax for it, as I would expect this to bind to a property Design on the DataContext).
```
<TextBlock IsVisible="{Binding !(Design.IsDesignMode)}"  Text="We are not in DesignMode" />
```

While on the topic of bindings I discovered that some properties of `Design` are exposed as attached properties and as such should be bindable. Perhaps we could expose `IsDesignMode` as a (read-only) attached property, then something like this should work:
```
<TextBlock IsVisible="{Binding $self.Design.IsDesignMode}"  Text="We are not in DesignMode" />
```

All in all I see three options to make my initial goal possible

1. Solving it only for precisely this scenario: Simply add a negated property to `Design`, i.e. `Design.IsNotDesignMode`. Than we can just use `x:Static` and be done with it.
2. Somehow add support for bindings to static classes/propperties.
3.  Expose `IsDesignMode` as a (read-only) attached property.

Does anyone have any other ideas that I have overlooked? If not which off the above proposals would you consider worthwhile? Which of them have a chance of being accepted? 1 and 3 seem to me as relatively easy to implement. I would be willing to do so.

**Comment by maxkatz6** [comment, marked as answer]
Worth a try:
```
{Binding !., Source={x:Static Design.IsDesignMode}}
```

**Reply by penenkel** [reply]
Well, darn, that actually works. Thanks
